### PR TITLE
Add support for LPC54018 IoT module board

### PR DESCRIPTION
**Pull request by mmahadevan108**
Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

### CODE REVIEW

**Comment by mmahadevan108** [comment]
cc @flit @brianesquilona @maclobdell

File: test/info.py [hunk 1]
```diff
@@ -168,6 +169,7 @@
     (   0x765,      'stm32f103xb_stm32l476rg_if',               'stm32f103xb_bl',   'ST-Nucleo-L476RG'                      ),
     (   0x796,      'stm32f103xb_stm32f429zi_if',               'stm32f103xb_bl',   'ST-Nucleo-F429ZI'                      ),
     (   0x816,      'stm32f103xb_stm32f746zg_if',               'stm32f103xb_bl',   'ST-Nucleo-F746ZG'                      ),
+    (   0x0235,     'lpc11u35_mini_iot_lpc54018_if',             None,               'MINI-54018'                              ),
```
Comment (flit) [inline test/info.py]: It's not a big deal, but the 3rd and 4th columns are misaligned by 1 character. It'd be best to fix it, but I'll still approve the PR without. 😄

Comment (mmahadevan108) [reply test/info.py]: Thank you. I have fixed this.

**Comment by mmahadevan108** [comment]
Can this be merged?

**Pull request merged**